From 84b7421d3c7f112c9501e564a86bd1bca4da4fc5 Mon Sep 17 00:00:00 2001
From: Kenneth Weiss <weiss27@llnl.gov>
Date: Wed, 5 Jul 2017 18:33:26 -0700
Subject: [PATCH] Patches gtest to not override a manually set
 GTEST_HAS_DEATH_TEST compiler define.

This compiler define can be manually set by the user
to disable gtest's death tests on unsupported platforms.
---
 .../googletest/include/gtest/internal/gtest-port.h                  | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/thirdparty_builtin/googletest-release-1.8.0/googletest/include/gtest/internal/gtest-port.h b/thirdparty_builtin/googletest-release-1.8.0/googletest/include/gtest/internal/gtest-port.h
index 0094ed5..15e9dcd 100755
--- a/thirdparty_builtin/googletest-release-1.8.0/googletest/include/gtest/internal/gtest-port.h
+++ b/thirdparty_builtin/googletest-release-1.8.0/googletest/include/gtest/internal/gtest-port.h
@@ -783,15 +783,17 @@ using ::std::tuple_size;
 #endif  // GTEST_HAS_STREAM_REDIRECTION
 
 // Determines whether to support death tests.
+#ifndef GTEST_HAS_DEATH_TEST
 // Google Test does not support death tests for VC 7.1 and earlier as
 // abort() in a VC 7.1 application compiled as GUI in debug config
 // pops up a dialog window that cannot be suppressed programmatically.
-#if (GTEST_OS_LINUX || GTEST_OS_CYGWIN || GTEST_OS_SOLARIS || \
+# if (GTEST_OS_LINUX || GTEST_OS_CYGWIN || GTEST_OS_SOLARIS || \
      (GTEST_OS_MAC && !GTEST_OS_IOS) || \
      (GTEST_OS_WINDOWS_DESKTOP && _MSC_VER >= 1400) || \
      GTEST_OS_WINDOWS_MINGW || GTEST_OS_AIX || GTEST_OS_HPUX || \
      GTEST_OS_OPENBSD || GTEST_OS_QNX || GTEST_OS_FREEBSD)
-# define GTEST_HAS_DEATH_TEST 1
+#  define GTEST_HAS_DEATH_TEST 1
+# endif
 #endif
 
 // We don't support MSVC 7.1 with exceptions disabled now.  Therefore
-- 
2.8.0

